### PR TITLE
fix(Combobox): move focus to options on keyboard navigation

### LINKED ISSUE #1001: [Bug]: Combobox options not announced by screen readers
### Environment

```bash
Developement/Production OS: Windows 10 19043.1110
Node version: 16.0.0
Package manager: pnpm@8.6.0
Radix Vue version: 1.0.0
Vue version: 3.0.0
Nuxt version: 3.0.0
Nuxt mode: universal
Nuxt target: server
CSS framework: tailwindcss@3.3.3
Client OS: Windows 10 19043.1110
Browser: Chrome 90.0.4430.212
```

### Link to minimal reproduction

https://www.radix-vue.com/components/combobox

### Steps to reproduce

1. Keep your screen reader on
2. Try navigating through the combobox options using arrow keys or home/end
3. None of the options are announced

### Describe the bug

None of the combobox options are announced by screen readers. This is due to the fact that the combobox component doesn't move focus to the option. It merely provides a visual feedback based on data attribute updates - but navigating with arrow keys doesn't move focus to the option.

### Expected behavior

Focus should be moved to the option that is being navigated through.

### Context & Screenshots (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by abh1nash**
This fixes #1001

### CODE REVIEW

File: packages/radix-vue/src/Combobox/ComboboxRoot.vue [hunk 1]
```diff
@@ -226,6 +227,13 @@ function scrollSelectedValueIntoView() {
     selectedElement.value.scrollIntoView({ block: 'nearest' })
 }
 
+function focusOnSelectedElement() {
+  // Find the highlighted element and focus
+  // This helps the screen readers to read the selected value
+  if (selectedElement.value instanceof Element && selectedElement.value.focus)
+    selectedElement.value.focus()
+}
+
```
Comment (zernonia) [inline packages/radix-vue/src/Combobox/ComboboxRoot.vue:236]: We don't need this focus function as [`aria-activedescendant`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-activedescendant) already handle this

File: packages/radix-vue/src/Combobox/ComboboxRoot.vue [hunk 2]
```diff
@@ -256,6 +265,9 @@ provideComboboxRootContext({
       selectedValue.value = filteredOptions.value[val === 'up' ? index - 1 : index + 1]
 
     scrollSelectedValueIntoView()
+    focusOnSelectedElement()
+
+    nextTick(() => inputElement.value?.focus({ preventScroll: true }))
```
Comment (zernonia) [inline packages/radix-vue/src/Combobox/ComboboxRoot.vue:270]: similarly this is not required

**Review by zernonia — CHANGES_REQUESTED** [review]
LGTM @abh1nash ! Thanks for the issue and PR!! 💚

Some minor comment and we can get this merge! 🚀

File: packages/radix-vue/src/Combobox/ComboboxRoot.vue (hunk 1 above)
Comment (abh1nash) [reply packages/radix-vue/src/Combobox/ComboboxRoot.vue:236]: I thought the same - but that doesn't work with all screen readers apparently.

File: packages/radix-vue/src/Combobox/ComboboxRoot.vue (hunk 2 above)
Comment (abh1nash) [reply packages/radix-vue/src/Combobox/ComboboxRoot.vue:270]: > similarly this is not required

Same here - focus announces the option, then this returns the focus back the input field - just as how one would expect a combobox to work.

File: packages/radix-vue/src/Combobox/ComboboxRoot.vue (hunk 1 above)
Comment (zernonia) [reply packages/radix-vue/src/Combobox/ComboboxRoot.vue:236]: Icic.. If that's the case then no prob 😁

**Review by zernonia — APPROVED** [review]

**Pull request merged**